### PR TITLE
[KYUUBI #2292] Unify kyuubi server plugin location

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Move `kyuubi-server-plugin` to `extension/kyuubi-server/`

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -82,7 +82,7 @@
         <module>kyuubi-hive-jdbc-shaded</module>
         <module>kyuubi-metrics</module>
         <module>kyuubi-server</module>
-        <module>kyuubi-server-plugin</module>
+        <module>extensions/kyuubi-server/kyuubi-server-plugin</module>
```
Comment (pan3793) [inline pom.xml:85]: please sort in alphabetically

Comment (ulysses-you) [reply pom.xml]: good catch

**Comment by cfmcgrady** [comment]
thanks, merge to master.

**Comment by cfmcgrady** [comment]
thanks, merge to master.

**Comment by ulysses-you** [comment]
thank you @cfmcgrady

File: pom.xml [hunk 2]
```diff
@@ -66,6 +66,7 @@
 
     <modules>
         <module>dev/kyuubi-codecov</module>
+        <module>extensions/kyuubi-server/kyuubi-server-plugin</module>
```
Comment (yaooqinn) [inline pom.xml:69]: extensions/server/kyuubi-server-plugin is better

Comment (cfmcgrady) [reply pom.xml:69]: let me raise a followup pr

**Pull request closed without merging**